src/www/ui/api/Controllers/UploadTreeController.php

```diff
@@ -0,0 +1,119 @@
+      $returnVal = new Info(404, "Edited License filter does not exist", InfoType::ERROR);
+    } else if ($scanFilter !== null && !$this->dbHelper->doesIdExist("license_ref", "rf_pk", $scanFilter)) {
+      $returnVal = new Info(404, "Scan License Filter does not exist", InfoType::ERROR);
+    } else if ($tagId !== null && !$this->dbHelper->doesIdExist("tag", "tag_pk", $tagId)) {
+      $returnVal = new Info(404, "Given Tag does not exist", InfoType::ERROR);
+    } else if ($openCBoxFilter !== null && ($openCBoxFilter !== 'true' && $openCBoxFilter !== 'false')) {
+      $returnVal = new Info(400, "openCBoxFilter must be a boolean value", InfoType::ERROR);
+    } else if ($flatten !== null && ($flatten !== 'true' && $flatten !== 'false')) {
+      $returnVal = new Info(400, "flatten must be a boolean value", InfoType::ERROR);
+    } else if ($sortDir != null && !($sortDir == "asc" || $sortDir == "desc")) {
+      $returnVal = new Info(400, "sortDirection must be asc or desc", InfoType::ERROR);
+    } else if ($page != null && !is_numeric($page)) {
+      $returnVal = new Info(400, "page must be an integer", InfoType::ERROR);
+    } else if ($show != null && $show != 'quick') {
+      $returnVal = new Info(400, "show must only be quick", InfoType::ERROR);
+    } else if ($limit != null && !is_numeric($limit)) {
+      $returnVal = new Info(400, "limit must be an integer", InfoType::ERROR);
+    } else {
+      $queryKeys = array_keys($query);
+      $allowedKeys = ['show', 'agentId', 'flatten', 'scanLicenseFilter', 'editedLicenseFilter', 'sort', 'page', 'limit', 'tagId', 'search', 'openCBoxFilter'];
+      $diff = array_diff($queryKeys, $allowedKeys);
+      if (count($diff) > 0) {
+        $returnVal = new Info(400, "Invalid query parameter(s) : " . implode(",", $diff), InfoType::ERROR);
+      }
+    }
+
+    if ($returnVal != null) {
+      return $response->withJson($returnVal->getArray(), $returnVal->getCode());
+    }
+
+    if ($show == 'quick') {
+      $uploadTreeId = $uploadDao->getFatItemId($uploadTreeId, $uploadId, $uploadDao->getUploadtreeTableName($uploadId));
+    }
+
+    $ajaxExplorerPlugin = $this->restHelper->getPlugin('ajax_explorer');
+    $symfonyRequest = new \Symfony\Component\HttpFoundation\Request();
+    $symfonyRequest->request->set('agentId', $agentId);
+    $symfonyRequest->request->set('tag', $tagId);
+    $symfonyRequest->request->set('item', $uploadTreeId);
+    $symfonyRequest->request->set('upload', $uploadId);
+    $symfonyRequest->request->set('fromRest', true);
+    $_GET['iSortingCols'] = "1";
+    $_GET['bSortable_0'] = "true";
+    $_GET['iSortCol_0'] = "0";
+    $_GET['sSortDir_0'] = $sortDir != null ? $sortDir : 'asc';
+    $_GET['iDisplayStart'] = $page;
+    $_GET['iDisplayLength'] = $limit;
+    $_GET["sSearch"] = $sSearch;
+    $_GET["conFilter"] = $editedFilter;
+    $_GET["scanFilter"] = $scanFilter;
```

Comment:
Instead of using magic variables here, can the corresponding locations be updated to use the `$request` argument in the `AjaxExplorer::handle()`?

It will make it very simple and safe.